src/PostgREST/ApiRequest.hs

```diff
@@ -124,13 +125,13 @@ userApiRequest schema req reqBody
   payload =
     case decodeContentType . fromMaybe "application/json" $ lookupHeader "content-type" of
       CTApplicationJSON ->
-          either Left (\val -> case ensureUniform (pluralize val) of
-            Nothing -> Left "All object keys must match"
-            Just json -> Right json) (JSON.eitherDecode reqBody)
+        (if BL.null reqBody && isTargetingProc
```

Comment:
Nice work with the bind, and also I love that `note` function, been wanting one like that for a while!

What if we reverse the direction of the bind? I think it makes the values appear to flow from right to left cleanly through the composition, plus putting the note first like a kind of comment:

```haskell
CTApplicationJSON ->
  note "All object keys must match" . ensureUniform . pluralize =<<
    if BL.null reqBody && isTargetingProc
       then Right emptyObject
       else JSON.eitherDecode reqBody
CTTextCSV ->
  note "All lines must have same number of fields" . ensureUniform . csvToJson =<<
    CSV.decodeByName reqBody
```